category/execution/monad/staking/fuzzer/staking_contract_machine.cpp

```diff
@@ -0,0 +1,2065 @@
+                    }
+                    return upper_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    if (upper_bound - lower_bound >= DUST_THRESHOLD) {
+                        return upper_bound - DUST_THRESHOLD;
+                    }
+                    return lower_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    if (upper_bound - lower_bound >= DUST_THRESHOLD + 1) {
+                        return lower_bound + DUST_THRESHOLD + 1;
+                    }
+                    return upper_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    if (upper_bound - lower_bound >= DUST_THRESHOLD + 1) {
+                        return upper_bound - (DUST_THRESHOLD + 1);
+                    }
+                    return lower_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    if (upper_bound - lower_bound >= DUST_THRESHOLD - 1) {
+                        return lower_bound + DUST_THRESHOLD - 1;
+                    }
+                    return upper_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    if (upper_bound - lower_bound >= DUST_THRESHOLD + 1) {
+                        return upper_bound - (DUST_THRESHOLD - 1);
+                    }
+                    return lower_bound;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    auto const x = gen_uint256();
+                    auto const m =
+                        1 +
+                        std::min(upper_bound - lower_bound, 2 * DUST_THRESHOLD);
+                    return m ? lower_bound + x % m : x;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    auto const x = gen_uint256();
+                    auto const m =
+                        1 +
+                        std::min(upper_bound - lower_bound, 2 * DUST_THRESHOLD);
+                    return m ? upper_bound - x % m : x;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    return lower_bound == upper_bound ? lower_bound
+                                                      : lower_bound + 1;
+                }),
+            Choice(
+                0.05,
+                [&](auto &) {
+                    return lower_bound == upper_bound ? upper_bound
+                                                      : upper_bound - 1;
+                }),
+            Choice(0.05, [&](auto &) { return lower_bound; }),
+            Choice(0.05, [&](auto &) { return upper_bound; }));
+    }
+
+    uint256_t StakingContractMachine::gen_stake(
```

Comment:
We should document what is going on in this function and `gen_bound_biased_uint256`.

Reply:
Done